### PR TITLE
Add wildcard IP address support to IsAuthorized

### PR DESCRIPTION
**Pull request by quantumagi**
See BUG 2970.

https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/2970/

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.RPC/RPCAuthorization.cs [hunk 1]
```diff
@@ -42,6 +42,10 @@ public bool IsAuthorized(IPAddress ip)
 
             if (this.AllowIp.Count == 0)
                 return true;
+
+            if (this.AllowIp.Any(i => i.ToString() == "0.0.0.0"))
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.RPC/RPCAuthorization.cs]: @quantumagi lets extend this to also facilitate the usage of ranges.

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.RPC/RPCAuthorization.cs]: Done.

**Comment by noescape00** [comment]
Sure, will review it

**Comment by dangershony** [comment]
@quantumagi please provide the Bitcoin link equivalent

File: src/Stratis.Bitcoin.Features.RPC/IPAddressBlock.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+namespace Stratis.Bitcoin.Features.RPC
+{
+    /// <summary>
+    /// The class of an object representing a block of ip addresses.
+    /// </summary>
+    public class IPAddressBlock
+    {
+        public int Mask { get; private set; }
+        public IPAddress Address { get; private set; }
```
Comment (noescape00) [inline src/Stratis.Bitcoin.Features.RPC/IPAddressBlock.cs]: add empty line

File: src/Stratis.Bitcoin.Features.RPC/IPAddressBlock.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+        public int FullMask => 8 * this.Address.GetAddressBytes().Length;
```
Comment (noescape00) [inline src/Stratis.Bitcoin.Features.RPC/IPAddressBlock.cs]: FullMaskLengthInBytes (or add a comment that this is a length of a mask in bytes)

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.RPC/IPAddressBlock.cs]: Renamed to `FullMaskLengthInBits`.

**Comment by quantumagi** [comment]
This fix relates to the solution put forward for https://github.com/bitcoin/bitcoin/issues/4894.

**Pull request merged**